django/utils/decorators.py

```diff
@@ -37,33 +37,47 @@ def _multi_decorate(decorators, method):
     else:
         decorators = [decorators]
 
-    def _wrapper(self, *args, **kwargs):
-        # bound_method has the signature that 'decorator' expects i.e. no
-        # 'self' argument, but it's a closure over self so it can call
-        # 'func'. Also, wrap method.__get__() in a function because new
-        # attributes can't be set on bound method objects, only on functions.
-        bound_method = wraps(method)(partial(method.__get__(self, type(self))))
+    if iscoroutinefunction(method):
+        # If the method is asynchronous, define an async wrapper.
+        async def _wrapper(self, *args, **kwargs):
+            # Bind the method to the instance.
+            bound_method = wraps(method)(partial(method.__get__(self, type(self))))
+            # Apply each decorator in order.
+            for dec in decorators:
+                bound_method = dec(bound_method)
+            # Await the decorated method.
+            return await bound_method(*args, **kwargs)
+
+        # Update the async wrapper with decorator attributes.
         for dec in decorators:
-            bound_method = dec(bound_method)
-        return bound_method(*args, **kwargs)
-
-    # Copy any attributes that a decorator adds to the function it decorates.
-    for dec in decorators:
-        _update_method_wrapper(_wrapper, dec)
-    # Preserve any existing attributes of 'method', including the name.
-    update_wrapper(_wrapper, method)
-    return _wrapper
+            _update_method_wrapper(_wrapper, dec)
+        # Update the async wrapper with the original method's attributes.
+        update_wrapper(_wrapper, method)
+        return _wrapper
+    else:
+        # If the method is synchronous, define a sync wrapper.
+        def _wrapper(self, *args, **kwargs):
+            # Bind the method to the instance.
+            bound_method = wraps(method)(partial(method.__get__(self, type(self))))
+            # Apply each decorator in order.
+            for dec in decorators:
+                bound_method = dec(bound_method)
+            # Call the decorated method.
+            return bound_method(*args, **kwargs)
+
+        # Update the sync wrapper with decorator attributes.
+        for dec in decorators:
+            _update_method_wrapper(_wrapper, dec)
+        # Update the sync wrapper with the original method's attributes.
+        update_wrapper(_wrapper, method)
+        return _wrapper
```

Comment:
You've still rewritten the comments from the existing code here. Please don't do that. 

Comments such as `# Call the decorated method.` followed by that exact code don't add anything.